staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -113,6 +133,7 @@ func (a CustomResourceDefinitionStorageStrategy) MatchCustomResourceDefinitionSt
 type customResourceValidator struct {
 	namespaceScoped bool
 	kind            schema.GroupVersionKind
+	validator       *validate.SchemaValidator
 }
 
 func (a customResourceValidator) Validate(ctx genericapirequest.Context, obj runtime.Object) field.ErrorList {
```

Comment:
is there are reason `obj` is not a `*unstructured.Unstructured` in the function signature? Do we need the type case below?